src/index.js

```diff
@@ -1,26 +1,24 @@
 import http from 'http'
 import { EventEmitter } from 'events'
 import { escapeRegExp } from './utils'
-import handler from './handler'
-
-const env = process.env.NODE_ENV || 'development'
-const metalStack = Symbol('metal:stack')
+import { metalStack } from './symbols'
+import baseHandler from './handlers/base'
 
 export default class Metal extends EventEmitter {
-  static createServer() {
-    const app = new Metal()
-    function appHandler(req, res, next) {
-      return appHandler.handle(req, res, next)
+  static createServer(...args) {
+    const app = new Metal(...args)
+    return (req, res, next) => {
```

Comment:
What's the benefit of this wrapper function? Can't we return `app.handle.bind(app)` or better app instance itself? (So we can listen on events, hooking into internals, etc)